kola/harness.go

```diff
@@ -418,16 +413,26 @@ func RunTest(t *register.Test, pltfrm string) error {
 		}
 	}
 
-	// run test
-	err = t.Run(tcluster)
-
 	// give some time for the remote journal to be flushed so it can be read
 	// before we run the deferred machine destruction
-	if err != nil {
-		time.Sleep(10 * time.Second)
-	}
+	defer func() {
+		if err != nil {
```

Comment:
Lets move this to the bottom of the switch below, seems silly to defer two functions when a single one will do.